ocw/lib/azure.py

```diff
@@ -100,126 +99,44 @@ def list_disks_by_resource_group(self, resource_group):
-        container_blobs = self.container_client(container.name).list_blobs()
-        for blob in container_blobs:
-            if (latest_modification > blob.last_modified):
-                latest_modification = blob.last_modified
-        if (self.older_than_min_age(latest_modification)):
-            self.log_info("Mark container for deletion {}", container.name)
-            if self.dry_run:
-                self.log_info("Deletion of boot diagnostic container {} skipped due to dry run mode", container.name)
-            else:
-                self.bs_client().delete_container(container.name)
-
-    def parse_image_name(self, img_name):
-        regexes = [
-            # SLES12-SP5-Azure.x86_64-0.9.1-SAP-BYOS-Build3.3.vhd
-            re.compile(r"""
-                       SLES
-                       (?P<version>\d+(-SP\d+)?)
-                       -Azure\.
-                       (?P<arch>[^-]+)
-                       -
-                       (?P<kiwi>\d+\.\d+\.\d+)
-                       -
-                       (?P<flavor>[-\w]+)
-                       -
-                       Build(?P<build>\d+\.\d+)
-                       \.vhd
-                       """,
-                       re.X),
-
-            # SLES15-SP2-BYOS.x86_64-0.9.3-Azure-Build1.10.vhd
-            # SLES15-SP2.x86_64-0.9.3-Azure-Basic-Build1.11.vhd
-            # SLES15-SP2-SAP-BYOS.x86_64-0.9.2-Azure-Build1.9.vhd
-            # SLES15-SP4-BYOS.x86_64-0.9.1-Azure-Build150400.2.103.vhd
-            re.compile(r"""
-                       SLES
-                       (?P<version>\d+(-SP\d+)?)
-                       (-(?P<type>[^\.]+))?\.
-                       (?P<arch>[^-]+)
-                       -
-                       (?P<kiwi>\d+\.\d+\.\d+)
-                       (-(?P<flavor>Azure[-\w]*))?
-                       -
-                       Build(\d+\.)?(?P<build>\d+\.\d+)
-                       \.vhd
-                       """,
-                       re.X)
-        ]
-        return self.parse_image_name_helper(img_name, regexes)
-
-    def cleanup_sle_images_container(self, keep_images):
-        container_client = self.container_client('sle-images')
-        for img in container_client.list_blobs():
-            m = self.parse_image_name(img.name)
-            if m:
-                self.log_dbg('Blob {} is candidate for deletion with build {} ', img.name, m['build'])
-
-                if img.name not in keep_images:
-                    self.log_info("Delete blob '{}'", img.name)
-                    if self.dry_run:
-                        self.log_info("Deletion of blob image {} skipped due to dry run mode", img.name)
-                    else:
-                        container_client.delete_blob(img.name, delete_snapshots="include")
+            if 'pcw_ignore' not in c['metadata']:
+                self.log_dbg('Found container {}', c.name)
+                container_blobs = self.container_client(c.name).list_blobs()
```

Comment:
Before we only looked for blobs in the `sle-images` container. Please also limit the behaviour here to only look for this container.

Reply:
done ( see  container_valid_for_cleanup )